src/plugins/intel_gpu/src/graph/common_utils/jit_term.hpp

```diff
@@ -220,17 +220,14 @@ inline JitTerm operator/(const JitTerm& lhs, const JitTerm& rhs) {
     return JitTerm{"(" + lhs.str() + " / " + rhs.str() + ")"};
 }
 inline JitTerm operator%(const JitTerm& lhs, const JitTerm& rhs) {
-    OPENVINO_ASSERT(rhs.str() != "0");
-    if (rhs.str() == "1" || rhs.str() == "-1") {
-        return JitTerm{"0"};
-    }
-
     if (is_number(lhs) && is_number(rhs)) {
         auto rhs_val = as_number<int64_t>(rhs);
         OPENVINO_ASSERT(rhs_val != 0, "Modulo by zero detected in operator%");
+        if (rhs_val == 1 || rhs_val == -1) {
```

Comment:
This will change the behavior when `rhs=="1"` and `lhs==!is_number`

Reply:
Fixed